benchmarks/graphics_pipeline/main.cpp

```diff
@@ -663,27 +688,36 @@ void ProjApp::SetupNoiseQuads()
         PPX_CHECKED_CALL(GetDevice()->CreatePipelineInterface(&piCreateInfo, &mNoiseQuads.pipelineInterface));
 
         mNoiseQuads.vertexBinding.AppendAttribute({"POSITION", 0, grfx::FORMAT_R32G32B32_FLOAT, 0, PPX_APPEND_OFFSET_ALIGNED, grfx::VERTEX_INPUT_RATE_VERTEX});
-
-        grfx::GraphicsPipelineCreateInfo2 gpCreateInfo  = {};
-        gpCreateInfo.VS                                 = {mVSNoise.Get(), "vsmain"};
-        gpCreateInfo.PS                                 = {mPSNoise.Get(), "psmain"};
-        gpCreateInfo.vertexInputState.bindingCount      = 1;
-        gpCreateInfo.vertexInputState.bindings[0]       = mNoiseQuads.vertexBinding;
-        gpCreateInfo.topology                           = grfx::PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP;
-        gpCreateInfo.polygonMode                        = grfx::POLYGON_MODE_FILL;
-        gpCreateInfo.cullMode                           = grfx::CULL_MODE_NONE;
-        gpCreateInfo.frontFace                          = grfx::FRONT_FACE_CW;
-        gpCreateInfo.depthReadEnable                    = true;
-        gpCreateInfo.depthWriteEnable                   = false;
-        gpCreateInfo.blendModes[0]                      = grfx::BLEND_MODE_NONE;
-        gpCreateInfo.outputState.renderTargetCount      = 1;
-        gpCreateInfo.outputState.renderTargetFormats[0] = GetSwapchain()->GetColorFormat();
-        gpCreateInfo.outputState.depthStencilFormat     = GetSwapchain()->GetDepthFormat();
-        gpCreateInfo.pPipelineInterface                 = mNoiseQuads.pipelineInterface;
-        PPX_CHECKED_CALL(GetDevice()->CreateGraphicsPipeline(&gpCreateInfo, &mNoiseQuads.pipeline));
     }
 }
 
+void ProjApp::CreateFullscreenQuadPipelines()
+{
+    grfx::GraphicsPipelineCreateInfo2 gpCreateInfo = {};
+    if (pSolidFullscreenQuads->GetValue()) {
+        gpCreateInfo.VS = {mVSColour.Get(), "vsmain"};
+        gpCreateInfo.PS = {mPSColour.Get(), "psmain"};
+    }
+    else {
+        gpCreateInfo.VS = {mVSNoise.Get(), "vsmain"};
+        gpCreateInfo.PS = {mPSNoise.Get(), "psmain"};
+    }
+    gpCreateInfo.vertexInputState.bindingCount      = 1;
+    gpCreateInfo.vertexInputState.bindings[0]       = mNoiseQuads.vertexBinding;
+    gpCreateInfo.topology                           = grfx::PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP;
+    gpCreateInfo.polygonMode                        = grfx::POLYGON_MODE_FILL;
+    gpCreateInfo.cullMode                           = grfx::CULL_MODE_NONE;
```

Comment:
any reason to disable the culling?

Reply:
Not particularly, I just copied code from `primitive_assembly` benchmark and made some modifications.

Reply:
i think backface culling would be better

Reply:
I just tried to change this to `CULL_MODE_BACK` and I can no longer see the fullscreen quad. I think it's getting culled? `CULL_MODE_FRONT` works though. Is that acceptable?

Reply:
NVM, I went with backface culling and I swapped the triangle vertices to make it point in the right direction.

Reply:
yeah, it is ok to change the vertices order (you can also change the winding to ccw). lgtm